From 38234c9299e208b773c392816910b8dc4534a173 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 12 Aug 2019 12:41:06 +0200
Subject: [PATCH] arm64: dt: fsl-imx8mq-tqma8mq: add cpu dvfs

this patch efectively enables overdrive mode - so we are able
to use higher CPU frequencies

- add operating points
- assign regulator to A53 cluster

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 .../boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi      | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
index 0198007f5267..b2b5a6e5f162 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
@@ -70,7 +70,7 @@
 		regulator-type = "voltage";
 		gpios = <&gpio1 6 GPIO_ACTIVE_HIGH>;
 		states = <900000 0x1 1000000 0x0>;
-		status = "disabled";
+		status = "okay";
 	};
 
 	thermal-zones {
@@ -80,6 +80,17 @@
 	};
 };
 
+&A53_0 {
+	operating-points = <
+		/* kHz    uV */
+		1500000 1000000
+		1300000 1000000
+		1000000 900000
+		800000  900000
+	>;
+	dc-supply = <&reg_tqma8mx_dvfs>;
+};
+
 &gpu {
 	status = "okay";
 };
